### PR TITLE
Simultaneous zip read write fix for inplace mode

### PR DESCRIPTION
**Pull request by yyakovlev**

### CODE REVIEW

File: src/main/java/com/bt/code/egress/write/FileCompleted.java [hunk 1]
```diff
@@ -13,7 +12,7 @@ public class FileCompleted {
     List<String> replacedLines;
 
     @FunctionalInterface
-    public interface Listener {
+    public interface FileListener {
```
Comment (kbabkin) [inline src/main/java/com/bt/code/egress/write/FileCompleted.java]: Can we rename it back to Listener?

File: src/main/java/com/bt/code/egress/write/FolderWriter.java [hunk 2]
```diff
@@ -49,6 +54,32 @@ public void onFileCompleted(FileCompleted fileCompleted) {
...
+        //Copy the resulting zip back in-place from temp dir
+        Path newZipAbsolutePath = getTempRoot().resolve(zipCompleted.getSourceZipRelativePath());
+
+        if (!Files.exists(newZipAbsolutePath)) {
+            //No changes to this zip file - skipping
+            return;
+        }
+
+        try {
+            Files.move(
```
Comment (kbabkin) [inline src/main/java/com/bt/code/egress/write/FolderWriter.java:68]: Would be good to log temp files creation and removal for better tracking

**Review by kbabkin — APPROVED** [review]

**Review by d57344 — APPROVED** [review]

File: src/main/java/com/bt/code/egress/write/FolderWriter.java (hunk 2 above)
Comment (yyakovlev) [reply src/main/java/com/bt/code/egress/write/FolderWriter.java:68]: Added

File: src/main/java/com/bt/code/egress/write/FileCompleted.java (hunk 1 above)
Comment (yyakovlev) [reply src/main/java/com/bt/code/egress/write/FileCompleted.java]: Renamed

**Pull request merged**